angular/projects/spark-angular/src/lib/components/sprk-dictionary/sprk-dictionary.component.spec.ts

```diff
@@ -44,71 +74,142 @@ describe('SprkAngularDictionaryComponent', () => {
-    component.ngOnInit();
-    fixture.detectChanges();
+    wrappedComponent.dictionaryType = 'striped';
+    wrappedFixture.detectChanges();
 
-    expect(element.classList.toString()).toContain(
+    expect(wrappedElement.classList.toString()).toContain(
       'sprk-c-Dictionary--striped',
     );
   });
 
-  it('should add data correctly', () => {
-    component.keyValuePairs = { key1: 'value1', key2: 'value2' };
-    fixture.detectChanges();
+  it('should add data correctly with deprecated Input', () => {
+    wrappedComponent.data = { key1: 'value1', key2: 'value2' };
+    wrappedFixture.detectChanges();
 
     expect(
-      element.getElementsByClassName('sprk-c-Dictionary__key').length,
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__key').length,
     ).toBe(2);
     expect(
-      element.getElementsByClassName('sprk-c-Dictionary__value').length,
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__value').length,
     ).toBe(2);
   });
 
-  it('should add data correctly with deprecated Input', () => {
-    component.data = { key1: 'value1', key2: 'value2' };
-    component.ngOnInit();
-    fixture.detectChanges();
+  it('should correctly respond to changes to data Input', () => {
+    wrappedComponent.data = { key1: 'value1', key2: 'value2' };
+    wrappedFixture.detectChanges();
+
+    expect(
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__key').length,
+    ).toBe(2);
+
+    wrappedComponent.data = { key1: 'value1', key2: 'value2', key3: 'value3' };
+    wrappedFixture.detectChanges();
+
+    expect(
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__key').length,
+    ).toBe(3);
+  });
+
+  it('should correctly respond to changes to keyValuePairs Input', () => {
+    wrappedComponent.keyValuePairs = { key1: 'value1', key2: 'value2' };
+    wrappedFixture.detectChanges();
 
     expect(
-      element.getElementsByClassName('sprk-c-Dictionary__key').length,
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__key').length,
     ).toBe(2);
+
+    wrappedComponent.keyValuePairs = {
+      key1: 'value1',
+      key2: 'value2',
+      key3: 'value3',
+    };
+    wrappedFixture.detectChanges();
+
+    expect(
+      wrappedElement.getElementsByClassName('sprk-c-Dictionary__key').length,
+    ).toBe(3);
+  });
+
+  it('should correctly respond to changes to dictionaryType Input', () => {
+    wrappedComponent.dictionaryType = '';
+    wrappedFixture.detectChanges();
+
+    expect(wrappedElement.classList.toString()).toBe('sprk-c-Dictionary');
+
+    wrappedComponent.dictionaryType = 'striped';
+    wrappedFixture.detectChanges();
+
+    expect(wrappedElement.classList.toString()).toContain(
+      'sprk-c-Dictionary--striped',
+    );
+  });
+
+  it('should prefer keyValuePairs over data', () => {
+    wrappedComponent.keyValuePairs = { key1: 'value1', key2: 'value2' };
+    wrappedComponent.data = { key1: 'value1', key2: 'value2', key3: 'value3' };
```

Comment:
is there a warning output if both are supplied? If both are there it means that a component is only partially moved over and [data] is being ignored